src/OpenColorIO/GpuShaderUtils.cpp

```diff
@@ -435,6 +435,14 @@ std::string GpuShaderText::declareVarStr(const std::string & name, float v)
     return floatDecl(name) + " = " + getFloatString(v, m_lang);
 }
 
+std::string GpuShaderText::vectorCompareExpression(const std::string& lhs, const std::string& op, const std::string& rhs)
+{
+    std::string ret = lhs + " " + op + " " + rhs;
+    if(m_lang == GPU_LANGUAGE_MSL_2_0)
+        ret = "any( " + ret + " )";
```

Comment:
one of the GPU test units produces following line in the MSL shader code and that causes compilation error in Metal.

```
float3 gamma = float3(0.790513813, 1.24223602, 0.828157365);
if ( gamma != float3(1., 1., 1.) )
{
```

To fix the compilation error, I added `any` that means if `any of the components are not equal`, then run the if-statement, otherwise jump over it or run the else-statement.